contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -35,94 +34,79 @@ interface ILinkAvailable {
 /// we could save a fair amount of gas and re-write this upkeep for use with Automation v2.0+,
 /// which has significantly different trust assumptions
 contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationCompatibleInterface {
-  using EnumerableMap for EnumerableMap.AddressToUintMap;
-
+  event BalanceUpdated(address indexed addr, uint256 oldBalance, uint256 newBalance);
   event FundsWithdrawn(uint256 amountWithdrawn, address payee);
-  event TopUpSucceeded(address indexed topUpAddress);
+  event MaxCheckUpdated(uint256 oldMaxCheck, uint256 newMaxCheck);
+  event MaxPerformUpdated(uint256 oldMaxPerform, uint256 newMaxPerform);
+  event MinWaitPeriodUpdated(uint256 s_minWaitPeriodSeconds, uint256 minWaitPeriodSeconds);
   event TopUpBlocked(address indexed topUpAddress);
+  event TopUpFailed(address indexed recipient);
+  event TopUpSucceeded(address indexed topUpAddress);
+  event TopUpUpdated(address indexed addr, uint256 oldTopUpAmount, uint256 newTopUpAmount);
   event WatchlistUpdated();
-  event MaxPerformUpdated(uint256 oldMaxPerform, uint256 newMaxPerform);
-  event MaxCheckUpdated(uint256 oldMaxCheck, uint256 newMaxCheck);
 
   error InvalidWatchList();
   error DuplicateAddress(address duplicate);
 
-  IERC20 private immutable LINK_TOKEN;
-  EnumerableMap.AddressToUintMap private s_watchList;
-  uint256 private s_topUpAmount;
-  uint32 private s_minWaitPeriodSeconds;
+  struct MonitoredAddress {
+    uint96 minBalance;
+    uint96 topUpAmount;
+    uint56 lastTopUpTimestamp;
+    bool isActive;
```

Comment:
nitpick - I don't think this field is that useful? Ex everywhere we ask `if (target.isActive) {...}` we could instead do `if (target.topUpAmount > 0) {...}` and I think it would have the same affect? I do not care at all if you don't change this, as the struct is still <32 bytes so I don't think there is much additional cost. Just calling it out.

Reply:
It's just an easy and cheap check, for example in places like 
```
      if (s_targets[targetAddress].isActive) {
        revert DuplicateAddress(addresses[idx]);
      }
```
At some point we can add methods to add/delete members that use this property as well.